posthog/models/historical_version.py

```diff
@@ -0,0 +1,44 @@
+from django.db import models
+from django.utils import timezone
+
+
+class HistoricalVersion(models.Model):
+    """
+    We don't _only_ store foreign references cos the referenced model might get deleted.
+    The history log should be relatively immutable
+
+    Everything in the log must have either a team id or an organization id
+    """
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["organization_id", "team_id", "name", "versioned_at"], name="unique_version"
+            ),
+            models.CheckConstraint(
+                check=models.Q(team_id__isnull=False) | models.Q(organization_id__isnull=False),
+                name="must_have_team_or_organization_id",
+            ),
+        ]
+
+    # JSON of the historical item
+    state = models.JSONField(null=False)
+    # e.g. feature_flags
+    name = models.fields.TextField(null=False)
+    # TODO will this only be create, update, or delete
+    action = models.fields.TextField(null=False)
```

Comment:
Similar as above - this needs work.

Reply:
And this has in practice stayed as only create, update, or delete. So can restrict it to a choice field